src/distributed-transaction/percolator.md

```diff
@@ -0,0 +1,153 @@
+# Percolator
+
+TiKV supports the distributed transaction, which is inspired by Google's paper about *Percolator*. We will introduce it and how we make use of it in TiKV in this section.
+
+## What is Percolator?
+
+*Percolator* is a system used for incremental processing on very large data set, built by Google. Support for distributed transactions is only a part of it. We will introduce Percolator briefly here. You can view the full paper [here](https://ai.google/research/pubs/pub36726#), and if you are very familiar to this paper, you can skip this section and go to read [Percolator in TiKV](#Percolator-in-TiKV)
+
+Percolator is built based on Google's BigTable. BigTable is a distributed storage system that supports single-row transactions, based on which Percolator implements distributed transactions in ACID snapshot-isolation semantics. A column `c` of Percolator is actually divided into following internal columns of BigTable:
+
+* `c:lock`
+* `c:write`
+* `c:data`
+* `c:notify`
+* `c:ack_O`
+
+Percolator also relies on a service named *timestamp oracle*. The timestamp oracle can produce timestamps in strictly increasing order. All read and write operations need to apply for timestamps from the timestamp oracle, and a timestamp came from timestamp oracle will be regarded as the logical time when the read/write operation happens.
+
+Percolator is multi-version, and a data item's version is represented by the timestamp that the transaction was committed.
```

Comment:
```suggestion
Percolator supports multi-version, and a data item's version is represented by the timestamp when the transaction was committed.
```

Reply:
I wanted to use "multi-version" as an adjective. If that's weird, how is it if I say "Percolator is a multi-version storage"?